pkg/controller/infinispan/resources/config/config_controller.go

```diff
@@ -152,3 +167,130 @@ func (c configResource) computeAndReconcileConfigMap(xsite *configuration.XSite)
+	err := r.reconcileOperatorConfigMap()
+	if err != nil {
+		r.Log.Error(err, "Error while computing and reconciling ConfigMap")
+		return reconcile.Result{Requeue: true}, nil
+	}
+	return reconcile.Result{}, nil
+}
+
+// reconcileOperatorConfigMap reconcile the operator ConfigMap
+// Operator configMap is unique for an operator install and placed in k8sutil.GetOperatorNamespace()/infinispan-operator-config
+func (r ReconcileInfinispanOperatorConfig) reconcileOperatorConfigMap() error {
+	config := &corev1.ConfigMap{}
+	operatorNs, err := getOperatorNamespace()
+	isRunningLocal := false
+	if err == k8sutil.ErrRunLocal {
+		isRunningLocal = true
+	}
+	if err != nil && !isRunningLocal {
+		return err
+	}
+	r.client.Get(ctx, types.NamespacedName{Namespace: operatorNs, Name: "infinispan-operator-config"}, config)
+	// Reconcile grafana
+	grafanaNs := config.Data["grafanaDashboardNameSpace"]
+	if grafanaNs != "" {
+		// detect the GrafanaDashboard resource type resourceExists on the cluster
+		resourceExists, err := r.Kube.IsGroupVersionSupported(grafanav1alpha1.SchemeGroupVersion.String(), grafanav1alpha1.GrafanaDashboardKind)
+		if err != nil {
+			r.Log.Error(err, "Error checking Grafana support")
+		}
+		if !resourceExists {
+			r.Log.Info("Grafana CRD not present - not installing dashboard CR")
+			return nil
+		}
+		infinispanDashboard := grafana.Dashboard(grafanaNs)
+		operatorDeployment := &appsv1.Deployment{}
+		if !isRunningLocal {
+			result, err := infinispan.LookupResource("infinispan-operator", operatorNs, operatorDeployment, r.client, r.Log)
+			if err != nil {
+				r.Log.Error(err, "There was a problem looking up the infinispan operator to add the dashboard as a secondary resource")
+				return err
+			}
+			if result != nil {
+				r.Log.Info("Infinispan operator resource not initialized, not installing Grafana Dashboard")
+				return nil
+			}
+			// result and err should be analyzed
+			if err = controllerutil.SetControllerReference(operatorDeployment, infinispanDashboard, r.scheme); err != nil {
+				r.Log.Error(err, "Unable to set Grafana Dashboard controller reference to Infinispan operator - skipping creation")
+				return nil
+			}
+		}
+		if _, err = controllerutil.CreateOrUpdate(context.Background(), r.client, infinispanDashboard, func() error {
+			// Don't do any merge
+			return nil
+		}); err != nil {
+			return err
+		}
+		return nil
+	}
+	return nil
+}
+
+// getOperatorNamespace returns the operator namespace or the WATCH_NAMESPACE
+// value if the operator is running outside the cluster.
+// "OSDK_FORCE_RUN_MODE"="local" must be set if running locally (not in the cluster)
+func getOperatorNamespace() (string, error) {
+	operatorNs, err := k8sutil.GetOperatorNamespace()
+	// This makes everything works even running outside the cluster
+	if err != nil && err == k8sutil.ErrRunLocal {
+		operatorNs = os.Getenv("WATCH_NAMESPACE")
```

Comment:
There is a function, called `k8sutil.GetWatchNamespace()` present. But what happens if this value will have multiple records (comma separated)

Reply:
This is mostly added for dev, I think we can just pickup the first ns in the list

Reply:
Not sure that will be actual for dev only, in upstream without OLM and several watching namespaces may take place potentially

Reply:
do you mean with operator running locally? Anyway I'll go for taking the first one in the list, until we will need any special policy